runelite-client/src/main/java/net/runelite/client/plugins/bosslogger/BossLoggerPanel.java

```diff
@@ -0,0 +1,437 @@
+		});
+
+		second.add(refresh);
+		second.add(clear);
+
+		title.add(first, BorderLayout.WEST);
+		title.add(second, BorderLayout.EAST);
+	}
+
+	// Wrapper for creating LootPanel
+	private JPanel createLootPanel(Tab tab)
+	{
+		// Grab Tab Data
+		ArrayList<LootEntry> data = bossLoggerPlugin.getData(tab.getName());
+
+		// Unique Items Info
+		ArrayList<UniqueItem> list = UniqueItem.getByActivityName(tab.getName());
+		Map<Integer, ArrayList<UniqueItem>> sets = UniqueItem.createPositionSetMap(list);
+
+		// Create & Return Loot Panel
+		lootPanel = new LootPanel(data, sets, itemManager);
+
+		return lootPanel;
+	}
+
+	private void showTabDisplay(Tab tab)
+	{
+		createTabPanel(tab);
+
+		this.revalidate();
+		this.repaint();
+	}
+
+	private void showLandingPage()
+	{
+		createLandingPanel();
+
+		this.revalidate();
+		this.repaint();
+	}
+
+	// Wrap the panel inside a scroll pane
+	private JScrollPane wrapContainer(JPanel container)
+	{
+		JPanel wrapped = new JPanel(new BorderLayout());
+		wrapped.add(container, BorderLayout.NORTH);
+		wrapped.setBackground(BACKGROUND_COLOR);
+
+		JScrollPane scroller = new JScrollPane(wrapped);
+		scroller.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+		scroller.getVerticalScrollBar().setPreferredSize(new Dimension(8, 0));
+		scroller.setBackground(BACKGROUND_COLOR);
+
+		return scroller;
+	}
+
+	// Updates panel for this tab name
+	void updateTab(String tabName)
+	{
+		// Change to tab of recently killed boss if on landing page
+		if (currentTab == null)
+		{
+			SwingUtilities.invokeLater(() -> showTabDisplay(Tab.getByName(tabName)));
+			return;
+		}
+
+		// only update the tab if they are looking at this boss tab
+		if (tabName.equals(currentTab.getName()))
+		{
+			// Reload data from file to ensure data and UI match
+			bossLoggerPlugin.loadTabData(currentTab);
+			// Grab LootPanel that needs to be updated
+			SwingUtilities.invokeLater(() -> lootPanel.updateRecords(bossLoggerPlugin.getData(tabName)));
+		}
+	}
+
+	void toggleTab(String tab, boolean flag)
+	{
+		// Only toggle tab if on landing page since the tabs are recreated each time
+		if (currentTab == null)
+			createLandingPanel();
+	}
+
+	// Refresh the Loot Panel with updated data (requests the data from file)
+	private void refreshLootPanel(LootPanel lootPanel, Tab tab)
+	{
+		// Refresh data for necessary tab
+		bossLoggerPlugin.loadTabData(tab);
+
+		// Recreate the loot panel
+		lootPanel.updateRecords(bossLoggerPlugin.getData(tab.getName()));
+
+		// Ensure changes are applied
+		this.revalidate();
+		this.repaint();
+	}
+
+	private void clearData(Tab tab)
+	{
+		if (lootPanel.getRecords().size() == 0)
+		{
+			JOptionPane.showMessageDialog(this.getRootPane(), "No data to remove!");
+			return;
+		}
+
+		int delete = JOptionPane.showConfirmDialog(this.getRootPane(), "<html>Are you sure you want to clear all data for this tab?<br/>There is no way to undo this action.</html>", "Warning", JOptionPane.YES_NO_OPTION);
```

Comment:
May be worth considering moving this, and any other HTML into either class level finals, or a HTML const file